packages/itwinui-css/src/table/base.scss

```diff
@@ -397,9 +394,14 @@
   position: relative;
   word-break: break-word;
 
+  .iui-table-body:not(:has(+ .iui-table-paginator)) > .iui-table-row:last-of-type & {
+    &:first-of-type {
+      border-end-start-radius: var(--iui-border-radius-1);
```

Comment:
Yes, the cell also needs the `border-radius` style. Not having it causes the border to become straight along the cell. Video (watch the bottom left corner of the table): 

https://github.com/user-attachments/assets/a52170a9-59fc-4f98-a494-46c29650f4cd

The rows also need the styling on them, since the expandable content rows do not have cells inside of them, and rely on the `border-radius` provided by the rows.